### PR TITLE
Fix species->parameter conversion during PEtab import

### PR DESCRIPTION
**Pull request by dweindl**
Fixed and added function `petab.sbml.constant_species_to_parameters` and tests to AMICI, as this will be removed from PEtab soon.

### CODE REVIEW

File: python/bin/amici_import_petab.py [hunk 1]
```diff
@@ -81,7 +84,7 @@ def show_model_info(sbml_model):
 
 
 def get_fixed_parameters(condition_file_name, sbml_model,
-                         constant_species_to_parameters=True):
+                         const_species_to_parameters=True):
```
Comment (yannikschaelte) [inline python/bin/amici_import_petab.py]: Parameter description in docstring?

File: python/bin/amici_import_petab.py [hunk 2]
```diff
@@ -138,6 +140,65 @@ def get_fixed_parameters(condition_file_name, sbml_model,
...
+
+    This can be used e.g. for setting up models with condition-specific
+    constant species for PEtab, since there it is not possible to specify
+    constant species in the condition table.
+
+    Arguments:
+        sbml_model: libsbml model instance
+
+    Returns:
+        species IDs that have been turned into constants
```
Comment (yannikschaelte) [inline python/bin/amici_import_petab.py]: list of ...

File: python/bin/amici_import_petab.py [hunk 3]
```diff
@@ -110,11 +113,10 @@ def get_fixed_parameters(condition_file_name, sbml_model,
...
 
-    if constant_species_to_parameters:
+    if const_species_to_parameters:
         # Turn species which are marked constant in the SBML model into
         # parameters
-        constant_species = \
-            petab.constant_species_to_parameters(sbml_model)
+        constant_species = constant_species_to_parameters(sbml_model)
 
         print("Constant species converted to parameters",
```
Comment (yannikschaelte) [inline python/bin/amici_import_petab.py]: since you added a logger ...

File: python/bin/amici_import_petab.py [hunk 4]
```diff
@@ -138,6 +140,65 @@ def get_fixed_parameters(condition_file_name, sbml_model,
...
+
+        if math.isnan(species.getInitialConcentration()):
+            logger.warning(
+                f"Ignoring {species.getId()} which has no initial "
+                "concentration. Amount conversion not yet implemented.")
+            continue
+
+        transformables.append(species.getId())
+
+    # Must not remove species while iterating over getListOfSpecies()
```
Comment (yannikschaelte) [inline python/bin/amici_import_petab.py]: ... and turn species to constant parameter?

File: python/sdist/setup.py [hunk 5]
```diff
@@ -185,7 +185,8 @@ def main():
                           'pkgconfig'],
         setup_requires=['setuptools>=40.6.3'],
         python_requires='>=3.6',
-        extras_require={'wurlitzer': ['wurlitzer']},
+        extras_require={'wurlitzer': ['wurlitzer'],
+                        'petab': ['petab>=0.0.0a15']},
```
Comment (yannikschaelte) [inline python/sdist/setup.py]: note: 15 is the current version on pypi (just to make sure)

File: python/bin/amici_import_petab.py (hunk 4 above)
Comment (dweindl) [reply python/bin/amici_import_petab.py]: not, simply not remove while iterating

**Pull request merged**